judge/contest_format/ioi16.py

```diff
@@ -0,0 +1,95 @@
+from django.db import connection
+from django.utils.translation import gettext_lazy
+
+from judge.contest_format.ioi import IOIContestFormat
+from judge.contest_format.registry import register_contest_format
+from judge.timezone import from_database_time
+
+
+@register_contest_format('ioi16')
+class IOI16ContestFormat(IOIContestFormat):
+    name = gettext_lazy('IOI16')
```

Comment:
I'd prefer renaming the old IOI format to "IOI (pre-2017)", `LegacyIOIContestFormat`, etc. I don't have a huge preference as whether to keep the format key as `ioi` and have this new one be different, or whether a migration should be written to update the format name.